pkg/projectfile/nameval.go

```diff
@@ -0,0 +1,53 @@
+package projectfile
+
+import (
+	"gopkg.in/yaml.v2"
+)
+
+type NameVal struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+func (nv *NameVal) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	data := make(map[string]interface{})
+	if err := unmarshal(&data); err != nil {
+		return err
+	}
+
+	if len(data) == 1 {
+		ssData, err := sanitizeMap(data)
```

Comment:
Can we rename this to "ensureStringMap()" and add comments to the function explaining that the reason for this function is to preserve yaml errors.